libphal/phal_dump.C

```diff
@@ -169,473 +219,574 @@ void writeSbeData(const std::filesystem::path& dumpPath, uint32_t reasonCode,
 }
 
 /**
- * @brief Initializes the PDBG and LIBEKB environments.
- *
- * @throw std::runtime_error Throws if either PDBG or LIBEKB initialization
- * fails.
+ * @brief Initializes PDBG backend with KERNEL mode and libekb
+ * Exceptions: PDBG_INIT_FAIL for any pdbg init related failure.
  */
 void initializePdbgLibEkb()
 {
+	// Initialize PDBG with KERNEL backend
 	pdbg::init(PDBG_BACKEND_KERNEL);
+
 	if (libekb_init()) {
 		log(level::ERROR, "libekb_init failed");
 		throw std::runtime_error("libekb initialization failed");
 	}
 }
 
 /**
- * @brief Probes a target (FSI or PIB) for the given processor target.
- * @param[in] proc The processor target to probe the FSI/PIB target for.
- * @param[in] targetType The type of target to probe ("fsi" or "pib").
- * @return Pointer to the probed pdbg target. If the target is not operational,
- *         the function throws an exception.
- *
- * @throw dumpError_t Throws if the target is not found or not operational.
- */
-struct pdbg_target* probeTarget(struct pdbg_target* proc,
-				const char* targetType)
-{
-	char path[16];
-	sprintf(path, "/proc%d/%s", pdbg_target_index(proc), targetType);
-	struct pdbg_target* target = pdbg_target_from_path(NULL, path);
-	if (!target || pdbg_target_probe(target) != PDBG_TARGET_ENABLED) {
-		log(level::ERROR, "Failed to get or probe %s target for (%s)",
-		    targetType, pdbg_target_path(proc));
-		throw dumpError_t(exception::PDBG_TARGET_NOT_OPERATIONAL);
-	}
-	return target;
-}
-
-/**
- * @brief Checks the state of the SBE on the given PIB target.
- * @param[in] pib The PIB target to check the SBE state of.
+ * @brief Checks the SBE state
  *
- * @throw sbeError_t Throws if the SBE state is 'FAILED' (indicating a dump
- *        has already been collected) or if there is a failure in reading the
- * SBE state.
+ * @param pib_fsi The pib or fsi target depending upon the chip
+ * @param sbeTypeId The chip type ID
  */
-void checkSbeState(struct pdbg_target* pib)
+void checkSbeState(struct pdbg_target* pib_fsi, const int sbeTypeId)
 {
 	enum sbe_state state;
-	if (sbe_get_state(pib, &state)) {
+
+	// If the SBE dump is already collected return error
+	auto rv = -1;
+	if (PROC_SBE_DUMP == sbeTypeId)
+		rv = sbe_get_state(pib_fsi, &state);
+	else if (ODYSSEY_SBE_DUMP == sbeTypeId)
+		rv = sbe_ody_get_state(pib_fsi, &state);
+
+	if (rv) {
 		log(level::ERROR, "Failed to read SBE state information (%s)",
-		    pdbg_target_path(pib));
+		    pdbg_target_path(pib_fsi));
 		throw sbeError_t(exception::SBE_STATE_READ_FAIL);
 	}
+
 	if (state == SBE_STATE_FAILED) {
 		log(level::ERROR,
-		    "Dump is already collected from the SBE on (%s)",
-		    pdbg_target_path(pib));
+		    "Dump is already collected from the "
```

Comment:
what is the change in this line?

Reply:
Due to clang format it splitted into two line LN261 & LN262.